yxi/axi-calyx/axi-combined-calyx.futil

```diff
@@ -174,7 +174,9 @@ component m_read_channel(
       // on the data we read from cocotb
       ref data_received = seq_mem_d1(32, 8, 64);
       is_rdy = std_reg(1);
-      ref curr_addr = std_reg(64); 
+      ref curr_addr = std_reg(64);
+      //need to increment this
+      ref base_addr = std_reg(64);
```

Comment:
Just providing a "dumb" reading of this line: why would one ever need to increment the *base* address? Going by its name, a *base* address is always fixed: it's baked into the "communication protocol" for an accelerator, and we never need to change it. We're adding other stuff to the base address to get a given access address, but the base address never changes.

I think I understand, though, the `base_addr` register here actually just _starts_ at the "base address" for the AXI interface. In fact, it holds the current "AXI-side" address we need to access.

That is, we imagine that there are two "sides" to every location in the abstract memory: the AXI-side address, which is what we will send out over the AXI bus, and the memory-side address, which we send into the `std_mem_d1` or whatever. These are related by a linear equation, like `axi_addr = mem_addr * WIDTH / 8 + BASE_ADDR` or something. But importantly, when `mem_addr == 0`, `axi_addr = BASE_ADDR`. Does this terminology make sense?

(If so, maybe it would be sensible to change the names of these registers to reflect this notion.)

Reply:
Yes! This is a question I asked myself recently and I think a name change is warranted. The terminology makes sense and this is correct. The incrementing ties into part of what you spoke on [here](https://github.com/calyxir/calyx/issues/1853#issuecomment-1902840303). Incrementing allows us to avoid doing expensive conversions by having the AXI-facing address "follow" along as transfers (making up a larger transaction) occur. This allows us to more easily transfer all of our data into our internal memories during the load part of our load -> compute -> store scheme.